### PR TITLE
Disallow interacting with the global track state in `Player` and `Editor`

### LINKED ISSUE #14816: Music controller pause button does not pause gameplay
The `Now Playing`-tab can be brought up in replays by pressing `F6`. Interacting with it doesn't seem to do anything except make the gameplay stutter. Might be worthwhile to hide the `Now Playing`-tab in replays, EDIT: Also seems to happen while spectating and inside the editor.

**video:** https://www.youtube.com/watch?v=u7hUjr70XLI
**osu!lazer version:** 2021.916.0-lazer

### LINKED ISSUE #24271: The Play/Pause hotkey works separately from the Play/Pause in Editor resulting in weird interactions with each other
### Type

Game behaviour

### Bug description

Have your Play/Pause hotkey from Audio Hotkeys set to something other than spacebar (like "C")
Go to the editor of any mapset
Observe that:

- Clicking "C" when map is playing in the Editor results in a stutter resulted in an attempt of pausing the track
- Clicking "C" when map is paused in the Editor results in only the Audio being played, without the map
- Clicking "Spacebar" when map is paused in the editor but with the Audio played by the technique above, results in map being played from the point where the audio is being played

### Screenshots or videos

https://github.com/ppy/osu/assets/82095491/1ba90651-e3e7-4905-b6cd-57a9fc928c2d

### Version

2023.717.0-lazer

### Logs

[input.log](https://github.com/ppy/osu/files/12069988/input.log)
[legacy-ipc.log](https://github.com/ppy/osu/files/12069989/legacy-ipc.log)
[network.log](https://github.com/ppy/osu/files/12069990/network.log)
[performance.log](https://github.com/ppy/osu/files/12069991/performance.log)
[runtime.log](https://github.com/ppy/osu/files/12069992/runtime.log)
[updater.log](https://github.com/ppy/osu/files/12069993/updater.log)
[database.log](https://github.com/ppy/osu/files/12069994/database.log)

### LINKED ISSUE #24385: Weird behavior when using play/pause shortcuts when loading map
### Type

Game behaviour

### Bug description

When pressing play/pause while a map is loading, hit objects may appear sooner vs when you normally load a map
Sometimes a completely different song may play instead

### Screenshots or videos

Normal behavior

https://github.com/ppy/osu/assets/66184829/94597156-040f-457a-812a-39ee99912526

Second song playing over the beatmaps song

https://github.com/ppy/osu/assets/66184829/352022aa-1cab-4b87-b28a-35319125c02a

Hit objects appear too soon

https://github.com/ppy/osu/assets/66184829/06a79258-be8d-413f-af02-5fe99038a3aa

### Version

2023.717.0-lazer

### Logs

[database.log](https://github.com/ppy/osu/files/12188645/database.log)
[input.log](https://github.com/ppy/osu/files/12188646/input.log)
[network.log](https://github.com/ppy/osu/files/12188647/network.log)
[performance.log](https://github.com/ppy/osu/files/12188648/performance.log)
[runtime.log](https://github.com/ppy/osu/files/12188649/runtime.log)
[updater.log](https://github.com/ppy/osu/files/12188650/updater.log)

### PR DESCRIPTION
**Pull request by peppy**
- Closes https://github.com/ppy/osu/issues/24271
- Closes https://github.com/ppy/osu/issues/14816
- Closes https://github.com/ppy/osu/issues/24385

This PR also renames `AllowTrackAdjustments` to more understandable `ApplyModTrackAdjustments`

### CODE REVIEW

**Comment by peppy** [comment]
Before merging this, we should make sure to understand and adjust for https://github.com/ppy/osu/issues/24385.

**Comment by alexpado** [comment]
Regarding #24385 wouldn't locking media key like you did here right after a beatmap started loading would be a fix ?

tbh doesn't feel right to be able to play around with the playback in a loading screen, but it might be a "me" issue

**Comment by peppy** [comment]
Please read before commenting to avoid it becoming spammy. This is already mentioned in https://github.com/ppy/osu/issues/24385#issuecomment-1655044634.

File: osu.Game/Screens/IOsuScreen.cs [hunk 1]
```diff
@@ -67,7 +67,13 @@ public interface IOsuScreen : IScreen
...
         /// A <see langword="null"/> value means that the parent screen's value of this setting will be used.
         /// </summary>
-        bool? AllowTrackAdjustments { get; }
+        bool? ApplyModTrackAdjustments { get; }
+
+        /// <summary>
+        /// Whether control of the global track should be allowed via the music controller / now playing overlay.
+        /// A <see langword="null"/> value means that the parent screen's value of this setting will be used.
+        /// </summary>
+        bool? AllowGlobalTrackControl { get; }
```
Comment (bdach) [inline osu.Game/Screens/IOsuScreen.cs:76]: I guess one pain point here is that we have two levels of "global track disable" now: one being global `Beatmap.Value.Disabled` which prevents changing track, and this which prevents pausing/resuming the current track. Not sure that's necessarily bad, maybe worth an xmldoc mention.

File: osu.Game/Overlays/MusicController.cs [hunk 2]
```diff
@@ -40,6 +40,11 @@ public partial class MusicController : CompositeDrawable
         /// </summary>
         public bool UserPauseRequested { get; private set; }
 
+        /// <summary>
+        /// Whether control of the global track should be allowed.
+        /// </summary>
+        public readonly BindableBool AllowTrackControl = new BindableBool(true);
```
Comment (bdach) [inline osu.Game/Overlays/MusicController.cs:46]: Also this is really about _user_ control of the global track. As the implementation shows below, screens are still able to manipulate the global track if they need, it's just not via user request. Dunno, probably a matter of xmldoc and/or rename at most again.

**Review by bdach — COMMENTED** [review]
As per https://github.com/ppy/osu/issues/24385#issuecomment-1656867489, this fixes at least one of the fail cases described in that issue. Not sure about the other because I can't repro, asked reporter for clarification.

Left a few comments but they're kinda for style points. I'd be honestly fine with merging this as-is and the only reason I'm not doing it right now is because of the other unexplained fail case.

**Comment by peppy** [comment]
@bdach if we're merging as-is, i'd want to move to `PlayerLoader` and `EditorLoader` to "cover" the other reported case. If you're happy with that then I'll just make the change.

**Comment by bdach** [comment]
No issue with that as far as I'm concerned

File: osu.Game/Screens/IOsuScreen.cs (hunk 1 above)
Comment (peppy) [reply osu.Game/Screens/IOsuScreen.cs:76]: Yeah, this one is over-arching (also disabled changing track) and should be used in most cases I guess.

The beatmap disable one seems fine to leave around for existing usages.

File: osu.Game/Overlays/MusicController.cs (hunk 2 above)
Comment (peppy) [reply osu.Game/Overlays/MusicController.cs]: Adjusted xmldoc.

**Review by bdach — APPROVED** [review]

**Pull request merged**